applications/app/controllers/MediaAtomEmbedController.scala

```diff
@@ -0,0 +1,66 @@
+package controllers
+
+import com.gu.contentapi.client.model.v1.ItemResponse
+import com.gu.contentatom.thrift.{Atom => AtomApi}
+import common._
+import model.Cached.{RevalidatableResult, WithoutRevalidationResult}
+import model._
+import play.api.mvc._
+
+import scala.concurrent.Future
+import contentapi.ContentApiClient
+import model.content.MediaAtom
+
+class MediaAtomEmbedController(contentApiClient: ContentApiClient) extends Controller with Logging with ExecutionContexts {
+
+  def render(id: String) = Action.async { implicit request =>
+    lookup(s"atom/media/$id") map {
+      case Left(model) => renderMediaAtom(model)
+      case Right(other) => renderOther(other)
+    }
+  }
+
+  def make(apiAtom: Option[AtomApi]): Option[MediaAtom] = {
+    apiAtom map {
+      MediaAtom.make
+    }
+  }
+
+
+
+  private def lookup(path: String)(implicit request: RequestHeader) = {
+    val edition = Edition(request)
+
+    log.info(s"Fetching media atom: $path for edition $edition")
+
+    val response: Future[ItemResponse] = contentApiClient.getResponse(contentApiClient.item(path, edition)
+
+    )
+
+    val result = response map { response =>
+
+      val modelOption = make(response.media)
+
+      modelOption match {
+        case Some(x) => Left(x)
+        case _ => Right(NotFound)
+      }
+    }
+
+    result recover convertApiExceptions
```

Comment:
You can probably chain everything here and save a few lines without sacrificing readibility. ie:
```
    response map { response =>
      make(response.media) match {
        case Some(x) => Left(x)
        case _ => Right(NotFound)
      }
    } recover convertApiExceptions
```